server/src/modules/admin/admin.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable, BadRequestException } from '@nestjs/common';
+import { StrategyService } from '../strategy/strategy.service';
+import { PerformanceService } from '../performance/performance.service';
+import { StartStrategyDto, StopStrategyDto } from './admin-strategy.dto';
+
+@Injectable()
+export class AdminService {
+  constructor(
+    private readonly strategyService: StrategyService,
+    private readonly performanceService: PerformanceService,
+  ) {}
+
+  async startStrategy(startStrategyDto: StartStrategyDto) {
+    const { strategyType, arbitrageParams, marketMakingParams, volumeParams } =
+      startStrategyDto;
+
+    if (strategyType === 'arbitrage' && arbitrageParams) {
+      return this.strategyService.startArbitrageStrategyForUser(
```

Comment:
`arbitrageParams` do not have `userId` and `clientId`, but this function call is for user to start arbitragy. What happens inside the function `startArbitrageStrategyForUser`, if the userId is not provided?

Reply:
It depends on what you need to do buy they will be used for tracking the strategy, the strategy key is made up of userid-clientid-strategytype think of it as an identifier for the strategy, just give it a random unique number, you can make the client id fix and increment the userid strating from 100 and going up or something like that.